### ISSUE TITLE
Bug: Generates invalid code due to inheritance

### ISSUE THREAD

**Issue by japgolly** [liancheng/scalafix-organize-imports#65]
If you have

```scala
class Codec {
  val Writer = 123
}
trait Settings {
  val codec = new Codec
}

object Server extends Settings
```

and then an import

```scala
import Server.codec.Writer
```

0.3.1-RC2 generates this

```scala
import com.blah.blah.Settings.codec.Writer
```

which doesn't compile. It should be `s/Settings/Server/`

```scala
import com.blah.blah.Server.codec.Writer
```

**Comment by liancheng** [comment]
Successfully reproduced this issue with the following patch:

```diff
diff --git a/input/src/main/scala/fix/Inheritance.scala b/input/src/main/scala/fix/Inheritance.scala
new file mode 100644
index 0000000..5e13f14
--- /dev/null
+++ b/input/src/main/scala/fix/Inheritance.scala
@@ -0,0 +1,9 @@
+/*
+rules = [OrganizeImports]
+OrganizeImports.expandRelative = true
+ */
+package fix
+
+import SomeObject.field.any
+
+object Inheritance
\ No newline at end of file
diff --git a/output/src/main/scala/fix/Inheritance.scala b/output/src/main/scala/fix/Inheritance.scala
new file mode 100644
index 0000000..15cba88
--- /dev/null
+++ b/output/src/main/scala/fix/Inheritance.scala
@@ -0,0 +1,5 @@
+package fix
+
+import fix.SomeObject.field.any
+
+object Inheritance
\ No newline at end of file
diff --git a/shared/src/main/scala/fix/Inheritance.scala b/shared/src/main/scala/fix/Inheritance.scala
new file mode 100644
index 0000000..1a42554
--- /dev/null
+++ b/shared/src/main/scala/fix/Inheritance.scala
@@ -0,0 +1,11 @@
+package fix
+
+class SomeClass {
+  val any: Any = ???
+}
+
+trait SomeTrait {
+  val field = new SomeClass
+}
+
+object SomeObject extends SomeTrait
```

I think we can fix this issue by expanding the top qualifier of an import expression, e.g., for `import a.b.C`, try to expand `a` instead of `a.b` by traversing ancestors of `a`.

**Comment by japgolly** [comment]
Nice work!

**Comment by liancheng** [comment]
Fixed this one and released 0.3.1-RC3.

**Comment by japgolly** [comment]
Yay! Thanks @liancheng !!
